net/server.go

```diff
@@ -30,19 +33,23 @@ var (
 
 // server implements the net gRPC server.
 type server struct {
-	sync.Mutex
 	net   *net
-	ps    *PubSub
 	opts  []grpc.DialOption
 	conns map[peer.ID]*grpc.ClientConn
+
+	ps     *pubsub.PubSub
+	topics map[thread.ID]*rpc.Topic
```

Comment:
Here are some mechanics for managing topics, which we need to do here now. This functionality used to be in a local `pubsub` package.